File: Desafio5/jogo_adivinhacao.java
import java.util.Scanner;
import java.util.Random;

public class JogoAdivinhacao {
    public static void main(String[] args) {
        Random random = new Random();
        Scanner scanner = new Scanner(System.in);

        int numeroSecreto = random.nextInt(100) + 1;
        int tentativa;
        int maxTentativas = 5;
        int contador = 0;

        System.out.println(" Jogo de Adivinhacao");
        System.out.println("Tente adivinhar um número entre 1 e 100.");
        System.out.println("Vc tem " + maxTentativas + " tentativas.");

        while (contador < maxTentativas) {
            System.out.print("\nTentativa " + (contador + 1) + ": ");
            tentativa = scanner.nextInt();

            if (tentativa < 1 || tentativa > 100) {
                System.out.println("Entrada invalida. Digite um numeero entre 1 e 100.");
                continue;
            }

            contador++;

            if (tentativa == numeroSecreto) {
                System.out.println("Parabéns! Você acertou o numero em " + contador + " tentativa(s).");
                break;
            } else if (tentativa < numeroSecreto) {
                System.out.println("O numero secreto eh maior que " + tentativa + ".");
            } else {
                System.out.println("O numero secreto eh menor que " + tentativa + ".");
            }
        }

        if (contador == maxTentativas) {
            System.out.println("\nVc perdeu. O numero era: " + numeroSecreto);
        }

        scanner.close();
    }
}

File: Desafio5/jogo_adivinhacao.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main() {
    int n_secreto, palpite;
    int tentativas = 0;
    const int maxTentativas = 5;

    srand(time(NULL));
    n_secreto = rand() % 100 + 1;

    printf("Jogo de Adivinhação\\n");
    printf("Tente adivinhar o número entre 1 e 100.\\nVocê tem %d tentativas.\\n", maxTentativas);

    while (tentativas < maxTentativas) {
        printf("Tentativa %d: ", tentativas + 1);
        scanf("%d", &palpite);

        if (palpite < 1 || palpite > 100) {
            printf("Entrada invalida Digite um numero entre 1 e 100.\\n");
            continue; 
        }

        tentativas++;

        if (palpite == n_secreto) {
            printf("Você acertou o numero em %d tentativa(s)!\\n", tentativas);
            break; 
        } else if (palpite < n_secreto) {
            printf("O numero e maior que %d.\\n", palpite);
        } else {
            printf("O numero e menor que %d.\\n", palpite);
        }
    }

    if (palpite != n_secreto) {
        printf("Voce perdeu O numero era %d.\\n", n_secreto);
    }

    return 0;
}
